src/main/java/org/mockito/internal/util/MockUtil.java

```diff
@@ -29,35 +29,26 @@
 public class MockUtil {
 
     private static final MockMaker defaultMockMaker = Plugins.getMockMaker();
-    private static final Map<Class<? extends MockMaker>, MockMaker> mockMakers;
-
-    static {
-        mockMakers = Collections.synchronizedMap(new WeakHashMap<>());
-        mockMakers.put(defaultMockMaker.getClass(), defaultMockMaker);
-    }
+    private static final Map<Class<? extends MockMaker>, MockMaker> mockMakers =
+            new ConcurrentHashMap<>(
+                    Collections.singletonMap(defaultMockMaker.getClass(), defaultMockMaker));
```

Comment:
I removed that `WeakHashMap` because I realized that it wouldn't remove anything anyway. (The class cannot be garbage collected when we are still holding the instance as the value.) However, I think not being able to unload a class loader because we have a reference to the mock maker is an acceptable limitation. I could not think of a realistic solution. (We could dynamically generate classes in the class loader which belongs to the mock maker to hold the reference, but I think that would be overkill. 😄)